### ISSUE TITLE
Set Ordinal type issues

### ISSUE THREAD

**Issue by apense** [nim-lang/Nim#3058]
Some of this is just a documentation issue. Where the manual says [``The set's basetype can only be an ordinal type``](http://nim-lang.org/docs/manual.html#types-set-type), I was thinking this would entail various ordinal types (`int8`, `int`, `byte`, `char`, etc.)
Exploring this with `typetraits` led to some interesting things. I'll give the example code and output here.

```nim
var s = {1'i8, 2}
echo type(s).name # set[int8]
```
```nim
var s = {1.int, 2}
echo type(s).name # set[range 0..65535(int)]
```
```nim
var s: set[int] # test.nim(40, 11) Error: set is too large
```
(this also happens with ``int32``)

```nim
var s = {1'i64, 2}
echo type(s).name # set[range 0..65535(int)]
```
```nim
var s: set[char]
echo type(s).name # set[char]
```
In short, it seems like downconversion to ``int16`` is happening, but, as long as this is supposed to be happening, the error messages and the documentation need some updating.

```nim
var s: set[uint]
echo type(s).name # test.nim(40, 11) Error: ordinal type expected
```
``uint`` and ``uint64`` aren't Ordinal, so that's a good error message.

**Comment by Araq** [comment]
Works all as intended, the manual needs to reflect reality.
